setup.py

```diff
@@ -35,31 +35,11 @@ def run(self):
         _build_py.run(self)
 
 
-install_requires = [
-    "ply>=3.9",  # See https://github.com/pyinstaller/pyinstaller/issues/1945
-    "configparser>=3.5.0",
-    "zc.lockfile>=1.2.1",
-    "future>=0.16.0",
-    "colorama>=0.3.9",
-    "configobj>=5.0.6",
-    "networkx>=2.1",
-    "gitpython>=2.1.8",
-    "setuptools>=34.0.0",
-    "nanotime>=0.5.2",
-    "pyasn1>=0.4.1",
-    "schema>=0.6.7",
-    "jsonpath-ng>=1.4.3",
-    "requests>=2.22.0",
-    "grandalf==0.6",
-    "asciimatics>=1.10.0",
-    "distro>=1.3.0",
-    "appdirs>=1.4.3",
-    "treelib>=1.5.5",
-    "inflect>=2.1.0",
-    "humanize>=0.5.1",
-    "dulwich>=0.19.11",
-    "ruamel.yaml>=0.15.91",
-]
+install_requires = []
+
+# Requirements to run dvc
+with open("requirements.txt") as fobj:
```

Comment:
Totally forgot to mention that our install_requires is not the same as requirements.txt. For example, in requirements, we have dependencies for all clouds supported, and install_requires doesn't have them, instead they are listed as optional packages. So you can't use the whole `requirements.txt` as `install_requires`.

Reply:
Let's see what OP @Suor can recommend.

Reply:
We can drop `requirements.txt` files and use:
```python
pip install -e .[tests]
pip install -e .[s3,gs]
pip install -e .[all]
```

Reply:
@Suor Could you provide examples of projects that have dropped requirements.txt? To make that happen, we'll need to rework our CI too, since both travis and appveyor use requirements.txt to setup local workspace.

Reply:
@efiop , I'd say is a common approach, check this out: https://caremad.io/posts/2013/07/setup-vs-requirement/

Reply:
Thanks @mroutis . Let's do that then.

@algomaster99 No, you'll just need to move tests/requirements.txt into an extra in setup.py. See https://github.com/iterative/dvc/blob/master/setup.py#L65 . We will have to do the same: just add "tests" to extras_require with a list of dependencies from tests/requirements.txt.

Reply:
@efiop looks like @mroutis got me covered. :+1:  My experience is the same: `requirements.txt` for projects, `setup.py` for libs. I also use `test-requirements.txt` in libs, but I started before `extras_require` appeared.